File: src/main/java/pl/igokaw/car/dto/CarFilterDto.java
package pl.igokaw.car.dto;

import lombok.Builder;
import pl.igokaw.car.enums.BodyTypeEnum;
import pl.igokaw.car.enums.TypeOfFuelEnum;

import java.math.BigDecimal;
import java.time.LocalDate;

@Builder
public record CarFilterDto (String brand, String model, BigDecimal engineCapacity,
                            BodyTypeEnum bodyType, TypeOfFuelEnum typeOfFuel, Integer productionYear,
                            LocalDate freeFrom, LocalDate freeTo){

}

File: src/main/resources/carrentalapp/src/app/cars/car-filter-dto.ts
import { BodyType } from "./enums/body-type";
import { TypeOfFuel } from "./enums/type-of-fuel";

export class CarFilterDto {
    brand!: string;
    model!: string;
    engineCapacity!: number;
    bodyType!: BodyType | null;
    typeOfFuel!: TypeOfFuel | null;
    productionYear!: number;
    freeFrom!: Date;
    freeTo!: Date;
}
